WIP WS API

tiagosiebler opened the pull request:
## Summary
<!-- Add a brief description of the pr: -->
- Bump node build version to LTS.
- Bump dependencies.
- Upgraded sign to use Web Crypto API.
- Introduced customSignMessageFn parameter (REST & WS) to inject custom sign method
  - This can be used by latency sensitive users to revert to Node's slightly faster createHmac sign method. See the fasterHmacSign example in examples/auth/ for a demonstration.
- Split CI tests to run separately (first public then private).


## Additional Information
<!-- Any additional information like breaking changes, dependencies added, screenshots, comparisons between new and old behavior, etc. -->

## Breaking Changes
This upgrade introduces the following breaking changes:
- REST Client constructor options simplified, see #111 for more details.
- 

## PR Checklist

As part of the PR, make sure you have:
- [ ] No breaking changes / documented all breaking changes clearly.
- [ ] Updated & checked that all tests pass.
- [ ] Updated the endpoint map (optional, if you know how).
- [ ] Increased the version number in the package.json <!-- Only if your changes need to be published to npm -->
- [ ] Checked `npm install` runs without issue.
- [ ] Included the package-lock.json, if it changed after npm install <!-- The version number should update, if you also updated the package.json version number -->
- [ ] Checked `npm run build` runs without issue.

JJ-Cro commented on src/rest-client.ts:
```diff
@@ -265,34 +260,6 @@ import {
 import BaseRestClient from './util/BaseRestClient';
 
 export class RestClient extends BaseRestClient {
-  /**
```
Was all of this supposed to be removed? Not sure where you moved it yet.

tiagosiebler replied:
Yes - in this case, the plan was to refactor the constructor signature to align with the other SDKs. Since these are just inherited from the extended `BaseRestClient`, and the old constructor was doing nothing but calling the parent, it's no longer needed in the REST client.

tiagosiebler commented on src/util/BaseRestClient.ts:
```diff
@@ -93,8 +100,14 @@ export default abstract class BaseRestClient {
       this.globalRequestOptions.headers = {};
     }
 
+    if ((options.market as any) === 'demo') {
```
The type is correct - this should technically not be possible anymore, unless a user isn't using typescript or an IDE that supports checking this and thinks passing `demo` (the old way) will work. Since it's quite a possible shock that the old way of saying `I want demo trading` has been refactored (breaking change), I thought it safer to throw a hard error for anyone still trying to use it.

tiagosiebler commented on examples/rest-private-trade.ts:
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Warnings are given further down unless the sell order portion of the example is uncommented. That being said, the sell part seems outdated (refers to old getSymbols() method that is no longer available). Would be worth updating the example separately.

tiagosiebler commented:
Published first public beta release, v3.0.0-beta.1, via #126.

```bash
npm install okx-api@beta
```

Changes since last beta:
- Implemented custom sign fn for Base REST client (it was missing).
